linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -1995,16 +1990,16 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
             js.Apply(newReceiver DOT transformMethodIdent(method), newArgs)
 
           def genDispatchApply(): js.Tree =
-            genCallHelper("dp_" + methodName, newReceiver :: newArgs: _*)
+            genCallHelper("dp_" + genName(methodName), newReceiver :: newArgs: _*)
 
           def genHijackedMethodApply(className: ClassName): js.Tree = {
-            val fullName = className + "__" + methodName
+            val fullName = genName(className) + "__" + genName(methodName)
```

Comment:
No helper in `JSGen` for this? (original comment seems unaddressed)

Reply:
Oops, I forgot that comment, indeed. I have now added the appropriate helpers (two overloads of `envField` in `JSGen`, and an additional overload of `envFieldDef` in `ClassEmitter`).